examples/documentation_examples/examples_test.go

```diff
@@ -37,30 +49,68 @@ func TestDocumentationExamples(t *testing.T) {
 
 	db := client.Database("documentation_examples")
 
-	documentation_examples.InsertExamples(t, db)
-	documentation_examples.QueryToplevelFieldsExamples(t, db)
-	documentation_examples.QueryEmbeddedDocumentsExamples(t, db)
-	documentation_examples.QueryArraysExamples(t, db)
-	documentation_examples.QueryArrayEmbeddedDocumentsExamples(t, db)
-	documentation_examples.QueryNullMissingFieldsExamples(t, db)
-	documentation_examples.ProjectionExamples(t, db)
-	documentation_examples.UpdateExamples(t, db)
-	documentation_examples.DeleteExamples(t, db)
-	documentation_examples.RunCommandExamples(t, db)
-	documentation_examples.IndexExamples(t, db)
-	documentation_examples.StableAPIExamples()
+	t.Run("InsertExamples", func(t *testing.T) {
+		documentation_examples.InsertExamples(t, db)
+	})
+
+	t.Run("QueryArraysExamples", func(t *testing.T) {
+		documentation_examples.QueryArraysExamples(t, db)
+	})
+
+	t.Run("ProjectionExamples", func(t *testing.T) {
+		documentation_examples.ProjectionExamples(t, db)
+	})
+
+	t.Run("UpdateExamples", func(t *testing.T) {
+		documentation_examples.UpdateExamples(t, db)
+	})
+
+	t.Run("DeleteExamples", func(t *testing.T) {
+		documentation_examples.DeleteExamples(t, db)
+	})
+
+	t.Run("RunCommandExamples", func(t *testing.T) {
+		documentation_examples.RunCommandExamples(t, db)
+	})
+
+	t.Run("IndexExamples", func(t *testing.T) {
+		documentation_examples.IndexExamples(t, db)
+	})
+
+	t.Run("StableAPExamples", func(t *testing.T) {
+		documentation_examples.StableAPIExamples()
+	})
+
+	t.Run("QueryToplevelFieldsExamples", func(t *testing.T) {
+		documentation_examples.QueryToplevelFieldsExamples(t, db)
+	})
+
+	t.Run("QueryEmbeddedDocumentsExamples", func(t *testing.T) {
+		documentation_examples.QueryEmbeddedDocumentsExamples(t, db)
+	})
+
+	t.Run("QueryArrayEmbeddedDocumentsExamples", func(t *testing.T) {
+		documentation_examples.QueryArrayEmbeddedDocumentsExamples(t, db)
+	})
+
+	t.Run("QueryNullMissingFieldssExamples", func(t *testing.T) {
+		documentation_examples.QueryNullMissingFieldsExamples(t, db)
+	})
+
+	mt := mtest.New(t)
 
 	// Because it uses RunCommand with an apiVersion, the strict count example can only be
 	// run on 5.0+ without auth. It also cannot be run on 6.0+ since the count command was
 	// added to API version 1 and no longer results in an error when strict is enabled.
-	ver, err := getServerVersion(ctx, client)
-	require.NoError(t, err, "getServerVersion error: %v", err)
-	auth := os.Getenv("AUTH") == "auth"
-	if testutil.CompareVersions(t, ver, "5.0") >= 0 && testutil.CompareVersions(t, ver, "6.0") < 0 && !auth {
-		documentation_examples.StableAPIStrictCountExample(t)
-	} else {
-		t.Log("skipping stable API strict count example")
-	}
+	mtOpts := mtest.NewOptions().MinServerVersion("5.0").MaxServerVersion("6.0").Auth(true)
+	mt.RunOpts("StableAPIStrictCountExample", mtOpts, func(t *mtest.T) {
+		documentation_examples.StableAPIStrictCountExample(t.T)
```

Comment:
It seems like the goal is to not run this test if auth is enabled. If that's the case I think you want `Auth(false)` instead.

However, since you updated the `StableAPIStrictCountExample` to use the `mtest.ClusterURI()`, it _should_ pass with auth and no auth, so you might be able to remove the `Auth()` limitation completely.